### PR TITLE
Feature/client search and dups

### PR DESCRIPTION
**Pull request by Mikmik28**
This pull request introduces a Ruby-based command-line application for processing client data from JSON sources. It includes core functionality for searching clients by name and identifying duplicate email addresses, along with a robust test suite and comprehensive documentation. Below are the most important changes grouped by theme.

### Core Functionality

* Added `Client` class to represent client data, with methods to split full names into first and last names and convert client attributes into a hash. (`lib/client.rb`)
* Implemented `ClientLoader` class to load client data from a JSON source (URL or file) and handle errors such as invalid URLs, missing files, or malformed JSON. (`lib/client_loader.rb`)
* Added `ClientSearcher` class to provide methods for searching clients by name (case-insensitive) and identifying duplicate email addresses. (`lib/client_searcher.rb`)

### Documentation and Setup

* Created `README.md` with detailed setup instructions, usage examples, assumptions, known limitations, and areas for future improvement. (`README.md`)
* Added a `Gemfile` to manage dependencies (`rspec` and `json`) for the application. (`Gemfile`)

### Testing

* Added RSpec tests for `Client` class to verify name splitting logic and edge cases. (`spec/client_spec.rb`)
* Added RSpec tests for `ClientLoader` class to validate loading from valid/invalid URLs and handling of malformed JSON. (`spec/client_loader_spec.rb`)
* Added RSpec tests for `ClientSearcher` class to ensure accurate search results and detection of duplicate emails. (`spec/client_searcher_spec.rb`)
* Configured RSpec with a basic setup in `spec_helper.rb`. (`spec/spec_helper.rb`)

### CODE REVIEW

File: lib/client_loader.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'json'
+require 'open-uri'
+require_relative 'client'
+
+class ClientLoader
+  def self.load(source)
+    raw_data = if source =~ URI::regexp
```
Comment (Mikmik28) [inline lib/client_loader.rb:7]: This line checks whether the value stored in the variable matches a valid URI
